### PR TITLE
Refactor curriculum import

### PR DESCRIPTION
**Pull request by LaMarseillaise**
These refactors simplify the `curriculum:update_content` rake task. The task is now split into import and verify steps (each their own task), and the actual importing work is done from the method `Lesson#import_content`.

One benefit of this refactor is that these tasks will be more modular, and therefore easier to maintain. Another is that there will not be less console output when running tests.

This update does not include migrations or require new environment variables.

### CODE REVIEW

File: spec/models/lesson_spec.rb [hunk 1]
```diff
@@ -59,4 +58,19 @@
       expect(lesson.position_in_section).to eql(2)
     end
   end
+
+  describe '#import' do
+    let(:github_lesson) { FactoryGirl.create(:lesson, url: "/README.md", content: nil) }
```
Comment (KevinMulhern) [inline spec/models/lesson_spec.rb]: We are in the process of moving away from factory girl in our tests, could you add `url: "/README.md"` and `content: nil` into the lesson double at the top of the and use the `lesson` double in your tests please.

File: app/models/lesson.rb [hunk 2]
```diff
@@ -21,6 +22,16 @@ def position_in_section
     ).count
   end
 
+  # Obtains lesson content from GitHub and saves it to the database
+  def import_content
```
Comment (KevinMulhern) [inline app/models/lesson.rb]: there's a lot going on in this method, can we refactor it out so it has a single responsibility. something like this please:

**Comment by KevinMulhern** [comment]
Hey @LaMarseillaise, thank you so much for opening this pull request. I left a couple of minor comments but everything else looks great. After you make the those changes I will merge. If you need me to clarify or help with anything at all, please don't hesitate to ask 😄

**Comment by LaMarseillaise** [comment]
I’ve made the requested changes. Definitely agree with the comments on `Lesson#import_content`.

**Comment by KevinMulhern** [comment]
Thanks @LaMarseillaise 
👍 merged

**Pull request merged**